pkg/build/builder/common.go

```diff
@@ -180,3 +155,54 @@ func execPostCommitHook(client DockerClient, postCommitSpec api.BuildPostCommitS
 		Stderr:       true,
 	})
 }
+
+func updateBuildRevision(build *api.Build, sourceInfo *git.SourceInfo) *api.SourceRevision {
+	if build.Spec.Revision != nil {
+		return build.Spec.Revision
+	}
+	return &api.SourceRevision{
+		Git: &api.GitSourceRevision{
+			Commit:  sourceInfo.CommitID,
+			Message: sourceInfo.Message,
+			Author: api.SourceControlUser{
+				Name:  sourceInfo.AuthorName,
+				Email: sourceInfo.AuthorEmail,
+			},
+			Committer: api.SourceControlUser{
+				Name:  sourceInfo.CommitterName,
+				Email: sourceInfo.CommitterEmail,
+			},
+		},
+	}
+}
+
+type buildDetails struct {
+	revision *api.SourceRevision
+	status   *api.BuildStatus
+}
```

Comment:
> the downside of the above is that every time i don't have a revision, which is a lot, i would end up initializing a pointer with no value to the function and it just seems inappropriate to me. plus i no longer look for a nil pointer, but for an empty field in the struct.

why can't you just pass nil?  all the args should be pointers and the updateStatus function should just ignore nil arg values.

the only problem w/ that is you can't set the value to nil intentionally, but do we have a need to do that currently?